### ISSUE TITLE
Issue with the new pathml version

### ISSUE THREAD

**Issue by mdu4003** [Dana-Farber-AIOS/pathml#261]
**Describe the bug**
We are having problems with running the pipeline using the new pathml version (2.0.1) . When we run the pipeline, we get the error below but the process completes anyway. When we look at the count matrix, we find that all the markers have the same intensity in each cell.

This is the output we get from slide.run:
```
WARNING:tensorflow:SavedModel saved prior to TF 2.5 detected when loading Keras model. Please ensure that you are saving the model with model.save() or tf.keras.models.save_model(), *NOT* tf.saved_model.save(). To confirm, there should be a file named "keras_metadata.pb" in the SavedModel directory.
2021-12-29 17:21:46.784209: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
WARNING:tensorflow:No training configuration found in save file, so the model was *not* compiled. Compile it manually.
2021-12-29 17:21:57.902989: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:176] None of the MLIR Optimization Passes are enabled (registered 2)
/Users/taniadiaz/miniforge3/envs/pathml-deepcell/lib/python3.8/site-packages/anndata/_core/anndata.py:120: ImplicitModificationWarning: Transforming to str index.
  warnings.warn("Transforming to str index.", ImplicitModificationWarning)
/Users/taniadiaz/miniforge3/envs/pathml-deepcell/lib/python3.8/site-packages/anndata/_io/h5ad.py:64: FutureWarning: The `force_dense` argument is deprecated. Use `as_dense` instead.
  warn(
Exception in thread "Thread-1" loci.formats.FormatException: Image plane too large. Only 2GB of data can be extracted at one time. You can work around the problem by opening the plane in tiles; for further details, see: https://docs.openmicroscopy.org/bio-formats/6.5.1/about/bug-reporting.html#common-issues-to-check
	at loci.formats.FormatReader.openBytes(FormatReader.java:872)
Caused by: java.lang.IllegalArgumentException: Invalid array size: 2900 x 0 x 1 x 4
	at loci.common.DataTools.safeMultiply32(DataTools.java:1281)
	at loci.common.DataTools.allocate(DataTools.java:1259)
	at loci.formats.FormatReader.openBytes(FormatReader.java:869)
Traceback (most recent call last):
  File "/Users/taniadiaz/miniforge3/envs/pathml-deepcell/lib/python3.8/site-packages/IPython/core/interactiveshell.py", line 3457, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-24-3ea9f30ca332>", line 1, in <module>
    slidedata.run(pipe, distributed = False, tile_size= (2900, 2900), tile_pad=True, overwrite_existing_tiles=True)
  File "/Users/taniadiaz/miniforge3/envs/pathml-deepcell/lib/python3.8/site-packages/pathml/core/slide_data.py", line 340, in run
    for tile in self.generate_tiles(
  File "/Users/taniadiaz/miniforge3/envs/pathml-deepcell/lib/python3.8/site-packages/pathml/core/slide_data.py", line 403, in generate_tiles
    for tile in self.slide.generate_tiles(shape, stride, pad, **kwargs):
  File "/Users/taniadiaz/miniforge3/envs/pathml-deepcell/lib/python3.8/site-packages/pathml/core/slide_backends.py", line 517, in generate_tiles
    tile_im = self.extract_region(
  File "/Users/taniadiaz/miniforge3/envs/pathml-deepcell/lib/python3.8/site-packages/pathml/core/slide_backends.py", line 368, in extract_region
    sample = reader.read(
  File "/Users/taniadiaz/miniforge3/envs/pathml-deepcell/lib/python3.8/site-packages/bioformats/formatreader.py", line 904, in read
    image = np.frombuffer(openBytes_func(index),dtype)
  File "/Users/taniadiaz/miniforge3/envs/pathml-deepcell/lib/python3.8/site-packages/bioformats/formatreader.py", line 784, in <lambda>
    openBytes_func = lambda x: self.rdr.openBytesXYWH(x, XYWH[0], XYWH[1], XYWH[2], XYWH[3])
  File "/Users/taniadiaz/miniforge3/envs/pathml-deepcell/lib/python3.8/site-packages/javabridge/jutil.py", line 961, in method
    result = call(self.o, name, sig, *args)
  File "/Users/taniadiaz/miniforge3/envs/pathml-deepcell/lib/python3.8/site-packages/javabridge/jutil.py", line 892, in call
    result = fn(*nice_args)
  File "/Users/taniadiaz/miniforge3/envs/pathml-deepcell/lib/python3.8/site-packages/javabridge/jutil.py", line 859, in fn
    raise JavaException(x)
javabridge.jutil.JavaException: Image plane too large. Only 2GB of data can be extracted at one time. You can work around the problem by opening the plane in tiles; for further details, see: https://docs.openmicroscopy.org/bio-formats/6.5.1/about/bug-reporting.html#common-issues-to-check
```
This is a screenshot of the count matrix:
<img width="710" alt="Screen Shot 2021-12-29 at 6 46 25 PM" src="https://user-images.githubusercontent.com/83671153/147711111-261219d2-82a1-4297-977d-2b5a899f73cc.png">

We ran the pipeline on the exact same slide but using pathml 1.0.1 and the process runs without any problems.
Thanks in advance

**Comment by jacobrosenthaldfci** [comment]
Hi Angeles, it looks like something went wrong and it's trying to read a slide of the image of shape (2900 x 0 x 1 x 4). 
Are you just calling `run()` with the default arguments, or are you passing specifying the tile size using something like `tile_size=256`?

What are the dimensions of the image?

**Comment by MohamedOmar2020** [comment]
Hi @jacobrosenthaldfci. This is the code I used with Angeles:

```
# load the fu** slide
slide = VectraSlide('data/MISI3590i_CTMA_117_AD_Scan2_Core[11,16,E]_[12878,56504]_component_data.tif')

slide.shape #(2900, 2900)

# Define a pipeline
pipe = Pipeline([
    CollapseRunsVectra(),
    SegmentMIF(model='mesmer', nuclear_channel=0, cytoplasm_channel=7, image_resolution=0.5),
    QuantifyMIF(segmentation_mask='cell_segmentation')
])
slide.run(pipe, distributed = False, tile_size= (2900, 2900), tile_pad=False, overwrite_existing_tiles=True)

# Get the count matrix
adata = slide.counts.to_memory()
adata
mat = adata.X
mat
```

**Comment by jacobrosenthaldfci** [comment]
There's a bug in the Bioformats backend, am looking into it more now. Thanks!

````python
bf_backend = BioFormatsBackend("data/MISI3590i_CTMA_117_AD_Scan2_Core[11,16,E]_[12878,56504]_component_data.tif")
im = bf_backend.extract_region(location = (0, 0), size = (2900, 2900))
n_channels = im.shape[3]
# all channels in extracted image are the same...
all([np.array_equal(im[:, :, :, 0, :], im[:, :, :, channel_ix, :]) for channel_ix in range(n_channels)])
````

**Comment by jacobrosenthaldfci** [comment]
Just as an update, this appears to be caused by a problem with the way the MISI image is formatted. Working on a fix now
@MohamedOmar2020 @mdu4003

**Comment by ryanccarelli** [comment]
Closed in #269
